File: vbnet/clock/Clock.vb
Structure Clock
    Private ReadOnly _clock As DateTime
    Sub New(hours As Integer, minutes As Integer)
        Dim t = (New DateTime(2, 1, 1, 0, 0, 0)).AddHours(hours).AddMinutes(minutes)
        _clock = New DateTime(2, 1, 1, t.Hour, t.Minute, 0)
    End Sub
    Function Add(minutesToAdd As Integer) As Clock
        Return New Clock(_clock.Hour, _clock.Minute + minutesToAdd)
    End Function
    Function Subtract(minutesToSubtract As Integer) As Clock
        Return New Clock(_clock.Hour, _clock.Minute - minutesToSubtract)
    End Function
    Public Overrides Function ToString() As String
        Return _clock.ToString("HH:mm")
    End Function
End Structure

File: php/clock/Clock.php
<?php

class Clock {
    private int $h;
    private int $m;

    function __construct(int $h, int $m = 0) {
        $this->h = $h % 24;
        $this->m = 0;
        $this->add($m);
    }

    public function add(int $minutes) {
        $this->h = ($this->h + intval(($this->m + $minutes) / 60)) % 24;
        $this->m = ($this->m + $minutes) % 60;
        if ($this->m < 0) {
            $this->h = ($this->h - 1) % 24;
            $this->m += 60;
        }
        if ($this->h < 0) $this->h += 24;
        return $this;
    }

    public function sub(int $minutes) {
        return $this->add(-$minutes);
    }

    public function __toString() {
        return sprintf("%02d:%02d", $this->h, $this->m);
    }
}

